connector/connect/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -47,6 +47,13 @@ message Relation {
 
     Unknown unknown = 999;
   }
+  // Optional. Every relation might have an alias.
```

Comment:
I don't have a strong opinion, but not sure which one is better. We can also follow catalyst, and add a new plan `SubqueryAlias(child, alias)`

Reply:
Given that this is already a message, also if we want to match DataFrame API, I think it makes more sense to have a `SubqueryAlias` in proto, and the call for `as(alias)` in the client will just create this `SubqueryAlias`, which is the same as existing DF implementation.

I have updated this PR accordingly.